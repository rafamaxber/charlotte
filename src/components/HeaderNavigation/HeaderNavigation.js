import React from 'react';
import styled from 'styled-components';

// TODO: Implement hamburguer menu for mobile

const LinkTexts = [
  { 
    text: 'The Queen city',
    link: '#',
    status: true,
  },
  {
    text: 'My Reservations',
    link: '#',
    status: false,
  },
  {
    text: 'Guide',
    link: '#',
    status: false,
  },
];

const Header = styled.header`
  @media (min-width: 570px) {
    width: 100%;
    max-width: 329px;
    height: 23px;
    justify-content: space-between;
    display: flex;
    left: 79px;
    top: 30px;
  }
  display: none;
  position: absolute;
`;

const StyledLink = styled.a`
  font-family: 'Heebo', sans-serif;
  position: relative;
  text-decoration: none;
  font-style: normal;
  font-weight: normal;
  line-height: normal;
  font-size: 15px;
  text-align: center;
  color: #FFFFFF;
  transition: 0.3s ease;
  cursor: ${ props => props.active ? 'default' : 'pointer' };;
  text-shadow: ${ props => props.active ? '0px 4px 4px rgba(0, 0, 0, 0.25)' : 'none' };
  &:after {
    content: "";
    display: block;
    height: 2px;
    background-color: #fff;
    bottom: -4px;
    position: absolute;
    left: 0;
    opacity: 0;
    width: 0;
    transition: .3s cubic-bezier(0.63, 0.23, 1, 1);
  }
  &:hover:after {
    width: 100% ;
    opacity: 1;
  }
`;

export default () => (
  <Header>
    {
      LinkTexts.map(item => (
        <StyledLink to={item.link} key={item.text} active={item.status}>
          { item.text }
        </StyledLink>
      ))
    }
  </Header>
);
