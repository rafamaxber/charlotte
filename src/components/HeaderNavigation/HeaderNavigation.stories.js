
import React from 'react';
import { storiesOf } from '@storybook/react';
import StoryRouter from 'storybook-router';
import styled from 'styled-components';
import Header from './HeaderNavigation';

const Stage = styled.div`
  background-color: #113875;
  width: 100%;
  height: 300px;
`;

storiesOf('Header', module)
  .addDecorator(StoryRouter())
  .add('Header', () => (
    <Stage>
      <Header></Header>
    </Stage>
  ));