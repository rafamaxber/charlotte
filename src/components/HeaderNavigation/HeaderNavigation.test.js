import React from 'react';
import renderer from 'react-test-renderer';
import HeaderNavigation from './HeaderNavigation';

describe('HeaderNavigation', () => {
  it('Should match with snapshot', () => {
    const component = renderer.create(
      <HeaderNavigation></HeaderNavigation>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
