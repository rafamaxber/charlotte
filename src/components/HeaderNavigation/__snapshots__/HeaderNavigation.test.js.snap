// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderNavigation Should match with snapshot 1`] = `
<header
  className="sc-bdVaJa fLsyZm"
>
  <a
    className="sc-bwzfXH ejqRSR"
    to="#"
  >
    The Queen city
  </a>
  <a
    className="sc-bwzfXH CoAOd"
    to="#"
  >
    My Reservations
  </a>
  <a
    className="sc-bwzfXH CoAOd"
    to="#"
  >
    Guide
  </a>
</header>
`;
